### DISCUSSION TITLE
DBus Daemon mode for Docker

### DISCUSSION THREAD

**Discussion by bublath** [bbernhard/signal-cli-rest-api#162] [Ideas]
I have an application that uses the Dbus Mode of signal-cli. While I know you don't like Dbus much, I see that you have a pretty sophisticated Docker creation that with RPC now also supports a kind of "daemon" mode.
While I really struggle providing my users a dockerized version (also due to my limited Docker know how), I would think that adding a Dbus MODE to your project should not be so complicated.
With 0.9.0 signal-cli daemon can be started without -u parameter and all the registration is done via Dbus as well, so a container that just runs a  signal-cli with "daemon --system" is what I'm looking for.
To make Dbus work however, I believe that
_/var/run/dbus/system_bus_socket_
would need to be exposed somehow (here my Docker know how definately is insufficent)

What do you think - could we add that (willing to help here) and prevent inventing the wheel again?

**Comment by bbernhard** [comment]
Hi,

if I understood you correctly, then you want to expose `signal-cli`'s DBUS interface directly via docker, right? In that case, I think it's probably better if you fork the repo and override the [entrypoint](https://github.com/bbernhard/signal-cli-rest-api/blob/master/Dockerfile#L176). I think that shouldn't be much work.

This project here creates a REST API interface around signal-cli's internal API, so signal-cli's internal API(s) aren't directly exposed to the user. While this obviously has some disadvantages (e.g we do not have every functionality of signal-cli exposed yet via APIs), this also has one big advantage: API stability and versioned APIs. So that's mainly the reason why this API wrapper exists :)
